incubator/hnc/pkg/kubectl/set.go

```diff
@@ -16,100 +16,153 @@ limitations under the License.
 package kubectl
 
 import (
+	"errors"
 	"fmt"
 	"os"
 
 	"github.com/spf13/cobra"
+
+	api "github.com/kubernetes-sigs/multi-tenancy/incubator/hnc/api/v1alpha1"
 )
 
+type flagStatus struct {
+	root                bool
+	parent              bool
+	parentString        string
+	requiredChild       bool
+	requiredChildArray  []string
+	optionalChild       bool
+	optionalChildString string
```

Comment:
Why do we need _both_ the `bool` and `[]string` versions? Don't we just need one?

Reply:
Yes I needed them separately because when `bool` values check the flag.Changed(values) is true or not. I think I can optimise it with one so I will try that approach once.